src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -33,58 +32,17 @@ public function __construct(ContainerInterface $container, LoggerInterface $logg
         parent::__construct($logger);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getController(Request $request)
-    {
-        $controller = parent::getController($request);
-
-        if (is_array($controller) && isset($controller[0]) && is_string($controller[0]) && $this->container->has($controller[0])) {
-            $controller[0] = $this->instantiateController($controller[0]);
```

Comment:
This does not belong here. I moved it to the base controller resolver. This is much more consistent and also makes array controller work where the class needs to be instantiated with old-school `new $class`.